### PR TITLE
Automated tests for habit CRUD

### LINKED ISSUE #8: Automated Tests for Habit CRUD
Write automated tests for habit creation, retrieval, update, and deletion APIs.

- [ ] Test habit creation
- [ ] Test fetching user habits
- [ ] Test updating a habit
- [ ] Test deleting a habit
- [ ] Verify users cannot access others’ habits

### PR DESCRIPTION
**Pull request by ZaidNayeb**
Automated Tests for Habit CRUD and Validator Middlewares 

Overview of Changes:

- [x]  Habit CRUD Tests & Habit Completion Tests
    - Covered all 8 core functions: createHabit, getHabit, getHabitsByDate, updateHabit, deleteHabit, reorderHabit, completeHabit, uncompleteHabit.
    - Implemented 52 test cases in total, including valid inputs, invalid inputs, edge cases, and error handling.

- [x] Validator Tests Refactoring
    - Split validateHabit tests into unit tests (mocked express-validator) and integration-style tests (real validator behavior).
    - Ensured all validator scenarios are covered: required fields, formatting, optional fields, and error message checks.
    - This improves readability, maintainability, and separates fast unit tests from slower integration-style tests.

- [x] User Validation Tests
    - Added 18 test cases for validateUser middleware covering all common scenarios: valid user data, missing fields, invalid formats, and error handling.

- [x]  Peer Review
    - Colleagues reviewed the code and provided feedback. All comments were addressed and resolved 

- [x] Reporter Review
   - Reported issues were verified and tested. The final implementation matches the requirements

closes #8

### CODE REVIEW

**Comment by AbdulQaharRahmani** [comment]
@Mahdi-Hasanzadeh @SakinaEbrahimi please review this

File: backend/src/_tests_/authController.test.js [hunk 1]
```diff
@@ -22,6 +22,27 @@ vi.mock('jsonwebtoken', () => ({
   },
 }));
 
+vi.mock('mongoose', () => ({
```
Comment (Mahdi-Hasanzadeh) [inline backend/src/_tests_/authController.test.js]: In this file, you should also mock the `RefreshToken` model.

**Review by Mahdi-Hasanzadeh — COMMENTED** [review]
Good job.
1: Pull main in your branch.
2: In the `Complete & Uncomplete habit` section, add tests to ensure that a user can only complete a habit for dates up to 7 days in the past, and cannot complete a habit for future dates.

File: backend/src/_tests_/authController.test.js (hunk 1 above)
Comment (ZaidNayeb) [reply backend/src/_tests_/authController.test.js]: I’ve added a mock for RefreshTokenModel in authController.test.js
let me know if you approve it.

File: backend/src/_tests_/controllers/habitsController.test.js [hunk 2]
```diff
@@ -0,0 +1,1166 @@
...
+  deleteHabit,
+  reorderHabits,
+  completeHabit,
+  uncompleteHabit,
+} from '../../controllers/habitsController';
+import { HabitModel } from '../../models/Habit';
+import { CategoryModel } from '../../models/Category';
+import { AppError } from '../../utils/error';
+import { HabitCompletionModel } from '../../models/habitCompletion';
+import { DateHelper } from '../../utils/date.js';
```
Comment (Mahdi-Hasanzadeh) [inline backend/src/_tests_/controllers/habitsController.test.js]: remove this line, we dont need it.
